### DISCUSSION TITLE
Suddenly I started getting 400 errors submitting pre-recorded MP3s

### DISCUSSION THREAD

**Discussion by fluen-ai** [deepgram/community#497] [General help]
### Which Deepgram product are you using?

Deepgram API

### Details

I'm transcribing pre-recorded videos, and to reduce overhead I only submit the audio track in MP3 format sicen I have it already separated from the video track, for other purposes. 

The setup worked normally for months until yesterday evening at around 11PM CET, all my requests are failing with a 400 error:
Error: "Bad Request: failed to process audio: corrupt or unsupported data"

This of course broke my whole workflow. I'm sending presigned S3 urls to where the MP3 is. I inspected the URLs, and they are perfectly working mp3 files like the ones I've been submitting for the past few months. What changed?
As a workaround, this morning I scrambled and made a patch to send presigned URLs to the whole MP4 video rendition instead, which seems to work, but this is not ideal, plus I may have audio only files I need to process, as well. 

### If you are making a request to the Deepgram API, what is the full Deepgram URL you are making a request to?

https://api.deepgram.com/v1/listen?callback=[redacted]&detect_language=true&model=whisper-large&diarize=true&punctuate=true&paragraphs=true&utterances=true&numerals=true&measurements=true

### If you are making a request to the Deepgram API and have a request ID, please paste it below:

f47e0ee8-fecf-4939-964e-54621e8a13ee

### If possible, please attach an example audio file to reproduce the issue.

https://github.com/deepgram/community/assets/133654036/05a1cd78-4ada-4154-a6d8-436b668a8a4c

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by fluen-ai** [comment]
Some additional details about the audio codec I'm using.
The mp3 files are rendered via AWS mediaconvert (using the original video as the source).
Here's the details:

Format: MPEG-4 AAC 
Bitrate control mode: CBR
Coding mode: 2.0 stereo
Sample rate: 48 kHz
Bitrate: 128 kbit/s

The issue appeared overnight, no changes or new deployments on my end. This audio format is the same I used for months.
Something must've happened on Deepgram's end at ~11pm CET of Dec 18th.

**Comment by fluen-ai** [comment]
Just in case someone else runs into a similar problem, I resolved the issue by changing the MP3 container settings in MediaConvert, from "MP4" to "RAW". This produces an mp3 file without unnecessary wraps, which Deepgram seems to successfully digest.
The Mp4 wrapper worked fine until yesterday, so I suppose something changed on DG's backend. Would be nice to be notified for breaking changes like this in the future. Thank you.

**Comment by jjmaldonis** [comment]
Hey @fluen-ai, glad you have a workaround that's working. I can't access the .mp4 you uploaded. Can you upload an mp3 file that is failing? You may need to zip the file first so GitHub allows the upload.

Is the mp4 -> mp3 conversion being done by AWS? I'd like to dig into this further. Currently, I can't reproduce the issue and I have not heard from other users experiencing similar issues. To my knowledge, nothing has changed on DG's backend that would affect how mp3s are processed.

**Reply by fluen-ai** [reply]
Here's a file sample that's not working:
[audio-x.mp3.zip](https://github.com/deepgram/community/files/13718239/audio-x.mp3.zip)

This file was generated via AWS MediaConvert from a user-submitted Mp4 video file.
If you're familiar with MediaConvert, here's the JSON portion of the encoding job:
```
{
        "CustomName": "mp3-audio",
        "Name": "File Group",
        "Outputs": [
          {
            "ContainerSettings": {
              "Container": "MP4",
              "Mp4Settings": {}
            },
            "AudioDescriptions": [
              {
                "AudioSourceName": "Audio Selector 1",
                "CodecSettings": {
                  "Codec": "AAC",
                  "AacSettings": {
                    "Bitrate": 128000,
                    "CodingMode": "CODING_MODE_2_0",
                    "SampleRate": 48000
                  }
                }
              }
            ],
            "Extension": "mp3"
          }
        ],
        "OutputGroupSettings": {
          "Type": "FILE_GROUP_SETTINGS",
          "FileGroupSettings": {
            "Destination": "s3://<MEDIABUCKET>/"
          }
        }
      }
```
As I said earlier, I had to swap the container from `MP4` to `RAW` and change the ContainerSettings to:
```
"CodecSettings": {
    "Codec": "AAC",
    "AacSettings": {
      "Bitrate": 128000,
      "CodingMode": "CODING_MODE_2_0",
      "SampleRate": 48000,
      "Specification": "MPEG4"
    }
  }
```
In order to make it work.

This is the CURL I used for testing:

```
curl \
  -X POST \
  -H "Authorization: Token <redacted>" \
  -H 'content-type: application/json' \
  -d '{"url":"<redacted_presigned_url_to_the_mp3_file_in_s3>"}' \
  "https://api.deepgram.com/v1/listen?detect_language=true&model=whisper-large&diarize=true&punctuate=true&paragraphs=true&utterances=true&numerals=true&measurements=true"
```
Since there were no changes on your end or my end, then the only reasonable explanation I have is that AWS somehow changed the MP4 container for audio files in a way that is not compatible anymore. Not sure.

**Reply by jjmaldonis** [reply]
Thanks for the file and all the info @fluen-ai. As an aside, we *rarely* get info as helpful as what you've provided, so thank you! Very helpful.

The file format of the mp3 file is very strange. Below are a few notes and inconsistencies:
- The linux command `file` analyzes a file to determine the file type. For the audio file you attached, `file` says it's an MP4 file. The exact output is `ISO Media, MP4 Base Media v1 [ISO 14496-12:2003]`.
- `ffprobe` is another utility for analyzing audio data. `ffprobe` identifies the audio as AAC audio: `Audio: aac (LC) (mp4a / 0x6134706D), 48000 Hz, stereo, fltp, 128 kb/s (default)`.
- When I use `ffmpeg` to "convert" the audio you attached into a different mp3 file, the re-encoded file is valid and Deepgram successfully transcribes it. I ran `ffmpeg -i audio-x.mp3 out.mp3` to re-encode the audio.

All of this points to the way MediaConvert is writing the audio data. In your MediaConvert JSON, can you try different parameters for the `CodecSettings` object? I'm not familiar with MediaConvert, but I'm guessing that changing some parameters will result in a more-valid mp3 file. You've essentially already done this so you may not want to make additional changes, but there may still be inconsistencies in the audio binary that could be worth fixing.
